CI is broken after ruff-0.5.0: E721

After ruff-0.5.0 release (https://github.com/astral-sh/ruff/releases/tag/0.5.0), our CI is broken due to E721 rule. 

See: https://github.com/huggingface/datasets/actions/runs/9707641618/job/26793170961?pr=6983
> src/datasets/features/features.py:844:12: E721 Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks

--- a/setup.py
+++ b/setup.py
@@ -215,7 +215,7 @@ METRICS_TESTS_REQUIRE = [
 TESTS_REQUIRE.extend(VISION_REQUIRE)
 TESTS_REQUIRE.extend(AUDIO_REQUIRE)
 
-QUALITY_REQUIRE = ["ruff>=0.3.0"]
+QUALITY_REQUIRE = ["ruff>=0.3.0,<0.5.0"]
 
 DOCS_REQUIRE = [
     # Might need to add doc-builder and some specific deps in the future

